pkg/state/state_skel.go

```diff
@@ -91,32 +92,103 @@ func (s *stateSkel) updateObj(obj *unstructured.Unstructured) error {
 	return nil
 }
 
+// When updating, not all resources need an updated ResourceVersion
+func needToUpdateResourceVersion(kind string) bool {
+	if kind == "SecurityContextConstraints" ||
+		kind == "Service" ||
+		kind == "ServiceMonitor" ||
+		kind == "Route" ||
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" {
+		return true
+	}
+	return false
+}
+
+func checkNestedFields(found bool, err error) {
+	if !found || err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+}
+
+func updateResourceVersion(req, found *unstructured.Unstructured) error {
+	kind := found.GetKind()
+
+	if needToUpdateResourceVersion(kind) {
```

Comment:
can you add a comment on why we need to update resource version ?

also i see you are just updating the field with the same value, is that ok ?

im probably lacking some k8s specific knowledge about `resourceVersion` so apologies if the questions seem obvious

Reply:
That's just the standard pattern for k8s afaik, but some OpenShift resources don't need it/are different API wise.

Reply:
From what i can tell from the below this is needed for implementing optimistic concurrency: 

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#concurrency-control-and-consistency